src/common/meta/src/version_reporter.rs

```diff
@@ -0,0 +1,200 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::env;
+use std::time::Duration;
+
+use common_runtime::{RepeatedTask, TaskFunction};
+use common_telemetry::info;
+use common_telemetry::tracing::log::warn;
+use once_cell::sync::Lazy;
+use reqwest::{Client, Response};
+use serde::{Deserialize, Serialize};
+
+use crate::error::{Error, Result};
+
+pub const VERSION_REPORT_URL: &str =
+    "https://api.greptime.cloud/opentelemetry/greptimedb-statistic";
+
+pub static VERSION_REPORT_INTERVAL: Lazy<Duration> = Lazy::new(|| Duration::from_secs(60 * 30));
+
+pub static VERSION_UUID_KEY: &'static str = "greptime_version_reporter_uuid";
+
+pub type VersionReportTask = RepeatedTask<Error>;
+
+#[derive(Serialize, Deserialize, Debug)]
+struct ReportData {
+    pub os: String,
+    pub version: String,
+    pub arch: String,
+    pub mode: String,
+    pub git_commit: String,
+    pub nodes: Option<i32>,
+    pub uuid: String,
+}
+
+pub enum Mode {
+    Distributed,
+    Standalone,
+}
+
+impl Mode {
+    pub fn to_string(&self) -> String {
+        match self {
+            Mode::Distributed => "distributed".to_string(),
+            Mode::Standalone => "standalone".to_string(),
+        }
+    }
+}
+
+#[async_trait::async_trait]
+pub trait Reporter {
+    async fn get_version(&self) -> String {
```

Comment:
Is it really necessary to be `async` here?

Reply:
not necessary. it should be change to sync fn.